components/salsa-2022-macros/src/input.rs

```diff
@@ -17,16 +17,36 @@ pub(crate) fn input(
     }
 }
 
-struct InputStruct(SalsaStruct);
+struct InputStruct(SalsaStruct<Self>);
 
 impl std::ops::Deref for InputStruct {
-    type Target = SalsaStruct;
+    type Target = SalsaStruct<Self>;
```

Comment:
Could someone explain to me why adding the `Self` parameter here is necessary? What does it enable that couldn't be done without the parameter present?

Reply:
Adding the `Self` parameter there is necessary as it needs to match the type of the newtype's tuple field, which is also a `SalsaStruct<Self>`. Aside from that, the functionality of the impl is the same (we're still providing access to the same fields and functions of `SalsaStruct`).